src/main/java/org/jenkinsci/plugins/argusnotifier/TagFactory.java

```diff
@@ -39,6 +40,20 @@ private TagFactory() {
                 .put(PROJECT.lower(), InvalidCharSwap.swapWithDash(projectName))
                 .build();
     }
+    
+    static Map<String, String> buildExtendedStatusTags(Jenkins jenkins, String projectName, String buildNumber,String commitId) {
+    	
+    	ImmutableMap.Builder<String,String> mapBuilder = ImmutableMap.<String, String>builder()
+                .putAll(hostTag(jenkins))
+                .put(PROJECT.lower(), InvalidCharSwap.swapWithDash(projectName));
+                
+        if (!commitId.isEmpty()) {
```

Comment:
Is this check needed?  `buildExtendedStatusTags` never gets called if `commitId` is empty.
https://github.com/jenkinsci/argus-notifier-plugin/blob/574c866442ffb6075056bd6968e18d50f344c447/src/main/java/org/jenkinsci/plugins/argusnotifier/BuildMetricFactory.java#L99-L101

Reply:
This is defensive check in TagFactory. But I agree it is duplicate check. I removed this check.